src/components/Accordion.tsx

```diff
@@ -0,0 +1,100 @@
+import { useId } from "@react-aria/utils";
+import { Dispatch, ReactNode, SetStateAction, useEffect, useState } from "react";
+import { useFocusRing } from "react-aria";
+import { Css, Icon, Palette, useTestIds } from "src/index";
+
+export interface AccordionProps {
+  title: ReactNode;
+  children: ReactNode;
+  disabled?: boolean;
+  defaultExpanded?: boolean;
+  size?: AccordionSize;
+  /** Adds a top border (enabled by default) */
+  topBorder?: boolean;
+  /** Adds a bottom border (disabled by default) */
+  bottomBorder?: boolean;
+  /**
+   * Used by AccordionList
+   * Allows multiple accordions to be expanded simultaneously (enabled by default)
+   */
+  index?: number;
+  setExpandedIndex?: Dispatch<SetStateAction<number | undefined>>;
+}
+
+export function Accordion(props: AccordionProps) {
+  const {
+    title,
+    children,
+    size,
+    disabled = false,
+    defaultExpanded = false,
+    topBorder = true,
+    bottomBorder = false,
+    index,
+    setExpandedIndex,
+  } = props;
+  const testIds = useTestIds(props, "accordion");
+  const id = useId();
+  const [expanded, setExpanded] = useState(defaultExpanded);
+  const { isFocusVisible, focusProps } = useFocusRing();
+
+  useEffect(() => { setExpanded(defaultExpanded); }, [defaultExpanded])
+
+  return (
+    <div
+      {...testIds.container}
+      css={{
+        ...Css.if(topBorder).bt.bGray400.if(bottomBorder && !(!expanded && isFocusVisible)).bb.$,
+        ...(size ? Css.wPx(accordionSizes[size]).$ : {}),
+      }}
+    >
+      <button
+        {...testIds.title}
+        {...focusProps}
+        aria-controls={id}
+        aria-expanded={expanded}
+        disabled={disabled}
+        css={{
+          ...Css.df.jcsb.gap2.aic.w100.p2.baseEm.outline("none").addIn(":hover", Css.bgGray100.$).$,
+          ...(disabled && Css.gray500.$),
+          ...(isFocusVisible && Css.boxShadow(`inset 0 0 0 2px ${Palette.LightBlue700}`).$),
+        }}
+        onClick={() => {
+          setExpanded(!expanded);
+          if (setExpandedIndex) setExpandedIndex(index);
+        }}
+      >
+        <span>{title}</span>
+        <span
+          css={{
+            transition: "transform 250ms linear",
+            transform: expanded ? "rotate(180deg)" : "rotate(0deg)",
+          }}
+        >
+          <Icon icon="chevronDown" />
+        </span>
+      </button>
+      <div
+        {...testIds.details}
+        id={id}
+        aria-hidden={!expanded}
+        css={{
+          // Use max-height for grow/shrink animation
+          ...Css.overflowHidden.maxhPx(1000).add("transition", "max-height 0.25s ease-in-out").$,
```

Comment:
I really think if we want to animate the height then we should calculate it via JS. Otherwise we are introducing some weird delays in the `allowMultipleExapanded === false` case.

Reply:
@bdow I added this condition `max-height ${expanded? "300ms" : "0"} ease-in-out` and it solves the delay issues (although it doesn't look as smooth when closing, but at least it keeps it just CSS)

Reply:
I could also add to calculate this only if `index` is set (meaning it comes from an `AccordionList`), leaving the single accordion with smooth animation while closing